cmd/reconciler/main.go

```diff
@@ -148,9 +292,103 @@ func main() {
 		}
 	}()
 
+	klog.Info("Starting manager")
+	defer func() {
+		// If the manager returned, there was either an error or a term/kill
+		// signal. So stop the other controllers, if not already stopped.
+		stopControllers()
+	}()
+	if err = mgr.Start(signalCtx); err != nil {
+		klog.Errorf("Error running manager: %v", err)
+		// os.Exit(1) does not run deferred functions so explicitly stopping the OC Agent exporter.
+		if err := oce.Stop(); err != nil {
+			klog.Errorf("Error stopping the OC Agent exporter: %v", err)
+		}
+		os.Exit(1)
+	}
+
+	// Wait for exit signal, if not already received.
+	// This avoids unnecessary restarts after the finalizer has completed.
+	<-signalCtx.Done()
+	klog.Info("All controllers exited")
+}
+
+// configureClients configures the set of clients used by Syncer, Remediator and Finalizer.
+//   - client.Client is used by the Finalizer to add/remove metadata.finalizer and set/remove finalizing condition
+//     It is also used by the Syncer to set the RSync status field.
+//   - DiscoveryInterface is used by the Syncer to build the scope of a resource.
+//   - Applier is used by the Remediator to correct any drift.
+//   - Supervisor is used by the Finalizer to finalize managed resources. It is also
+//     used by the Syncer to apply/prune declared resources.
+func configureClients(scope declared.Scope) (client.Client, discovery.DiscoveryInterface, reconcile.Applier, applier.Supervisor, error) {
+	reconcileTimeoutDuration, err := timeStringToDuration(*reconcileTimeout)
+	if err != nil {
+		return nil, nil, nil, nil, fmt.Errorf("reconcileTimeout format error: %v", err)
+	}
+
+	// Get a config to talk to the apiserver.
+	apiServerTimeoutDuration, err := time.ParseDuration(*apiServerTimeout)
```

Comment:
Why isn't `timeStringToDuration` used here?

Do we need to check whether it's non-negative? I see this being used for parsing other timeout strings but the check wasn't there

Reply:
Might want to avoid using package variables in this helper and instead pass them into the function pre-parsed and pre-validated.